chrisj117/CS-472-GROUP-2-PROJECT
# Backend authentication completed

**_WHAT DOES THIS PR DO?_**
- This PR completes user authentication use-case. It contain the following functionalities:

- [X] Request password reset
- [X] Reset password using registered email
- [X] Custom redirects for frontend and mobile app (incase it is need in the near future)

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- #92 
- #118 

**_HOW DO I TEST OUT THIS PR?_**

  - Check out this branch locally, `cd` to `api` and activate your virtual environment. Get a the lastest `.env` file from the project's discord page, then run `pip install -r requirements.txt` and `python manage.py runserver`.
  - Go to `https://localhost:8000` to use swagger for testing the endpoints:

<img width="1428" alt="Screen Shot 2023-11-23 at 10 10 41 PM" src="https://github.com/chrisj117/CS-472-GROUP-2-PROJECT/assets/15224992/262a0b09-a63c-45a8-bc27-003670599cb5">

- If you have  registered user account, you can use the `request-reset-email` endpoint

>  {
>   "email": "user@example.com",
>   "redirect_url": "string"
> }
Enter your registered email address. Delete the "redirect_url" part

- You will receive a password reset email, click the link and copy the UIDB64 and TOKEN from the URL  to `password-rest-complete` endpoint

> {
>   "password": "string",
>   "token": "string",
>   "uidb64": "string"
> }
